synchronizer/l1_live_block_ranges.go

```diff
@@ -5,23 +5,26 @@ import (
 	"fmt"
 )
 
-type liveBlockRangeItem struct {
+type liveBlockRangeItem[T any] struct {
 	blockRange blockRange
+	tag        T
```

Comment:
What is tag?

Reply:
Tag is a generic field to store any kind of data as extra information, you can use to store related information. 
It's a common field for Microsoft languages as VBA, C#,etc... that allow to add some extra information to already exisiting objects